odxtools/audience.py

```diff
@@ -35,7 +38,10 @@ def from_et(et_element, doc_frags: List[OdxDocFragment]) -> "AdditionalAudience"
     def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
         return {self.odx_id: self}
 
-    def _resolve_references(self, odxlinks: OdxLinkDatabase) -> None:
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
+        pass
+
+    def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
         pass
```

Comment:
This block seems to repeat everywhere, maybe move it to a common parent class ?

```
    def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
        return {self.odx_id: self}

    def _resolve_references(self, odxlinks: OdxLinkDatabase) -> None:
    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
        pass

    def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
        pass
```

Reply:
I'm not really sure if this is a good idea: having these methods around should reduce copy-and-paste mistakes because these functions are more visible, and also the `_build_odxlinks()` sample implementation of above snipped is only valid for classes which do not have any complex sub-objects but which an ID.

Reply:
to reduce copy-paste mistakes simply make a unit test that:
* checks that every ID attribute in sample odx XML end up registered in the odx link database.
* checks resolve gets called for each ID-REF in the odx xml (ny providing a mocked link database, and checking the call count)

Reply:
that's exactly the problem: while I would *love* to have such extensive test coverage, we currently still have a very long way to go to get there ...